proto/v1/rollout_service.proto

```diff
@@ -101,9 +117,406 @@ message Plan {
   // The system-assigned, unique identifier for a resource.
   string uid = 2 [(google.api.field_behavior) = OUTPUT_ONLY];
 
+  // The resource name of the review associated with this plan.
+  // Format: projects/{project}/reviews/{review}
+  string review = 3;
+
+  string title = 4;
+  string description = 5;
+
+  repeated Step steps = 6;
+
+  // FIXME(d/xz): support spec with deployment config
+  message Step {
+    repeated Spec specs = 1;
+  }
+
+  message Spec {
+    // earliest_allowed_time the earliest execution time of the change.
+    google.protobuf.Timestamp earliest_allowed_time = 4;
+
+    oneof config {
+      CreateDatabaseConfig create_database_config = 1;
+      ChangeDatabaseConfig change_database_config = 2;
+      RestoreConfig restore_config = 3;
+    }
+  }
+
+  message CreateDatabaseConfig {
+    // The resource name of the instance on which the database is created.
+    // Format: instances/{instance}
+    string target = 1;
+    // The name of the database to create.
+    string database = 2;
+    // table is the name of the table, if it is not empty, Bytebase should create a table after creating the database.
+    // For example, in MongoDB, it only creates the database when we first store data in that database.
+    string table = 3;
+    // character_set is the character set of the database.
+    string character_set = 4;
+    // collation is the collation of the database.
+    string collation = 5;
+    // cluster is the cluster of the database. This is only applicable to ClickHouse for "ON CLUSTER <<cluster>>".
+    string cluster = 6;
+    // owner is the owner of the database. This is only applicable to Postgres for "WITH OWNER <<owner>>".
+    string owner = 7;
+    // backup is the resource name of the backup.
+    // FIXME: backup v1 API is not ready yet, write the format here when it's ready.
+    string backup = 8;
+    // labels of the database.
+    map<string, string> labels = 9;
+  }
+
+  message ChangeDatabaseConfig {
+    // The resource name of the target.
+    // Format: projects/{project}/logicalDatabases/{ldb1}.
```

Comment:
unless it's a convention, I usually frown upon using camelCase, and would choose all lower case instead. projects/{project}/logical-databases/{ldb1}.

Reply:
It is. https://google.aip.dev/122#:~:text=Collection%20identifiers%20must%20be%20in%20camelCase.